input/delimiter/delimiter.go

```diff
@@ -0,0 +1,139 @@
+package delimiter
+
+import (
+	"bytes"
+	"errors"
+	"strconv"
+	"sync"
+)
+
+const (
+	LenBuffEnd = ":"
+	ValBuffEnd = ";"
+	NoResult   = false
+)
+
+var (
+	err        error
+	mutex      *sync.Mutex = &sync.Mutex{}
+	brokenErr              = errors.New("broken")
+	lenIncErr              = errors.New("length-buffer-incomplete")
+	lenInvErr              = errors.New("length-buffer-invalid-byte")
+	lenConvErr             = errors.New("length-buffer-conversion-error")
+	valIncErr              = errors.New("value-buffer-incomplete")
+)
+
+// A Delimiter detects when message lines start.
+type Delimiter struct {
```

Comment:
This needs a better name than just Delimiter. It's following a particular scheme, i.e. "netstrings".